Poll and Survey XBlock

mhoeber opened the pull request.

mhoeber commented:
@catong   poll and survey topics are mostly complete, there will be a few more items to add.

mhoeber commented:
reminders to self:

- Add full support lines to main tools table
- Add includes in OLX and open source guide
- change log for B&R, open source, and OLX

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+*********
+
+You can embed a poll in your course. A poll prompts learners to answer. A
+learner sees the poll and selects an appropriate answer.
```
This wording is a bit awkward. How about something like: "You can include polls in your course to gather learners' opinions on various questions."

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+Poll Tool
+###################
+
+This section describes how to embed polls in your course. 
```
"embed" -> "include"?

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+
+You can embed a poll in your course. A poll prompts learners to answer. A
+learner sees the poll and selects an appropriate answer.
+
```
Is there a reason we are not reusing the screenshot/example from the old "Poll" doc? ("All things being equal, should the government use public funds to invest in museums rather than sports arenas?") Just wondering why we seemed to be reverting to a rather simplistic and possibly unrealistic example.

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+learner sees the poll and selects an appropriate answer.
+
+.. image:: ../../../shared/building_and_running_chapters/Images/poll_tool.png
+    :alt: A poll as asking the learner's favorite color
```
Alt text needs correcting and more detail including the poll questions.

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+    :alt: A poll as asking the learner's favorite color
+    :width: 400
+
+When learners submit answers, they see results of the poll.
```
"When" -> "After" 
"After learners choose their answer to the poll question, they see the poll results that have been gathered at this time.

catong replied:
Hm, I just read below that this is a setting. "After learners choose their answers to the poll question, they see the poll results that have been gathered at this time, unless the poll has been configured to hide results."

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+When learners submit answers, they see results of the poll.
+
+.. image:: ../../../shared/building_and_running_chapters/Images/poll_with_results.png
+    :alt: A poll as asking the learner's favorite color
```
Fix alt text

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+======================================
+Enable the Poll Tool in Studio
+======================================
+
```
I suggest consistent wording with other similar procedures, so that don't introduce a lot of minor variants to these phrases. Better for translation, etc.

catong replied:
"Enable Content Libraries" is one example.

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+
+#. From the **Settings** menu, select **Advanced Settings**.
+
+#. In the **Advanced Module List** field, place your cursor between the braces,
```
"between the supplied pair of square brackets"

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+    :alt: Advanced modules setting for poll
+    :width: 400
+
+#. At the bottom of the page, select **Save Changes**.
```
I'd strike "At the bottom of the page". "Select **Save Changes**.

catong replied:
I think several of these procedures have the step result:
"Studio reformats the policy key you just entered to indent it on a new line."

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+Enable the Poll Tool
+*******************************************
+
+Before you can add a poll to your course, you must enable it in Studio or
```
"enable it" -> "enable the poll tool"

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+
+  <course advanced_modules="[&quot;survey&quot;, 
+      &quot;poll&quot;]" display_name="Sample Course" 
+      start="2014-01-01T00:00:00Z">
```
Better to have a more current seeming example -- I'd suggest changing the year in the example to 2015.

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+Add a Poll in edX Studio
+***************************
+
+Ensure you enable the poll tool before you add the component.
```
"Ensure you enable" -> "You must enable".
OR: "Before you can add a poll component in your course, you must enable the poll tool."
(And include link to the "enable" topics)

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+#. In the **Question/Prompt** field, enter text that learners see above the
+   poll. You can use Markdown in this field.
+
+#. In the **Feedback** field, enter text that learners see after submitting a
```
"after submitting" -> "after they submit responses."

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+
+#. In the **Private Results** field, to hide poll results from learners,
+   select **True**. If you leave the default value, **False**, learners see
+   poll results after submitting answers.
```
"after submitting answers." -> "after they submit responses."

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+   select **True**. If you leave the default value, **False**, learners see
+   poll results after submitting answers.
+
+#. In the **Maximum Submissions** field, to allow learners to submit answers
```
"answers" sounds odd in the context of a poll. I suggest "responses", as above.
"to submit responses more than once."

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+   poll results after submitting answers.
+
+#. In the **Maximum Submissions** field, to allow learners to submit answers
+   more than once, change the value. Enter **0** to allow unlimited
```
"...change the value to the number of times you want to allow learners to submit responses."

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+   submissions.
+
+   .. note:: 
+    If you allow learners to submit answers more than once, you should set
```
"answers" -> "responses"

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+
+   .. note:: 
+    If you allow learners to submit answers more than once, you should set
+    **Private Results** to **True**.
```
Why?

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+   #. In each **Answer** field, enter the answer text that learners see.
+      
+   #. You must enter either text or an image path, or both, for each answer.
+      To use an image, us the :ref:`Studio URL <File URLs>` for the image.
```
To "enter" an image, "use" (typo "us" -> "use")

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+   #. You must enter either text or an image path, or both, for each answer.
+      To use an image, us the :ref:`Studio URL <File URLs>` for the image.
+
+   #. If you use an image, you must enter alternative text in the **Image
```
Perhaps expand slightly: "you must enter useful alternative text in the **Image alternative text" field for non-sighted users.
**Please check the UI field name -- should be "alternative text" not "alternate text".

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+      To use an image, us the :ref:`Studio URL <File URLs>` for the image.
+
+   #. If you use an image, you must enter alternative text in the **Image
+      alternate text** field.
```
Change UI field name to match when/if it is corrected.

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+   #. If you use an image, you must enter alternative text in the **Image
+      alternate text** field.
+
+   #. The poll template contains three answer fields. To add answers, select
```
Should this be presented before the previous 2 steps? Some of these numbered steps should perhaps be informational intro text before getting into the procedure steps?

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+
+To add a poll XBlock in OLX, you create the ``poll`` element. You can embed
+the ``poll`` element in the ``vertical`` element, or you can create the
+``poll`` element as a stand-alone file that you reference in the vertical.
```
No hyphen, one word "standalone"

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+the ``poll`` element in the ``vertical`` element, or you can create the
+``poll`` element as a stand-alone file that you reference in the vertical.
+
+The following example shows the OLX definition for a poll with two questions.
```
Does it show 2 questions? I think there's only one question.

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+  "/>
+
+==========================
+poll Element Attributes
```
is "poll" lowercase here because it is an element name? Looks odd, but OK if consistent with other similar sections.

mhoeber replied:
yes, element is lc

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+poll Element Attributes
+==========================
+
+The following table describes the attribute of the ``poll`` element.
```
Missing "s" in "attributes" (plural)

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+     * - ``display_name``
+       - The display name for the poll.
+     * - ``question``
+       - The prompt for the poll.
```
I thought there could be more than one question in a poll? Does this question refer to the main prompt for the poll? If it does not refer to the main question but instead to questions within the poll, "question text" might be more understandable than "prompt".

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+       - The number of times a learner can submit poll answers.  Use ``0`` to
+         allow unlimited submissions. If you use a value other than ``1``, set
+         ``private_results`` to ``true``.
+     * - ``answers``
```
Same question as above -- I don't understand the reason behind this recommendation.

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+     * - ``max_submissions``
+       - The number of times a learner can submit poll answers.  Use ``0`` to
+         allow unlimited submissions. If you use a value other than ``1``, set
+         ``private_results`` to ``true``.
```
Same question as above -- I don't understand the reason behind this recommendation.

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+Editing Published Polls
+***************************
+
+Do not publish a poll until you have completed and tested it. You want to
```
You want to -> "You should"

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+***************************
+
+Do not publish a poll until you have completed and tested it. You want to
+avoid changing a poll after learners have begun using it.
```
"begun using it" -> "begun to use it."

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+Do not publish a poll until you have completed and tested it. You want to
+avoid changing a poll after learners have begun using it.
+
+If you have to edit a poll after learners have submitted answers:
```
"if you have to" -> "If you must"
"have submitted answers:" -> "have submitted answers, follow these guidelines."

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+
+If you have to edit a poll after learners have submitted answers:
+
+* If you edit the value an answer, previous submissions are
```
"the value an answer" - > "the value of an answer"

catong replied:
"previous submissions are associated with..."
Would be clearer to state something like: "previously correct submissions might no longer be correct, and vice versa."

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+
+* If you edit the value an answer, previous submissions are
+  associated with the new answer value. This change can cause you
+  to get an inaccurate picture of the responses.
```
"can cause you to get" - > "can result in"

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+  associated with the new answer value. This change can cause you
+  to get an inaccurate picture of the responses.
+
+* If you change the poll so that previous submissions are invalid, those
```
Not quite clear what this means -- does this mean that there is some option to "make previous submissions invalid" when the poll is changed, or does it just mean that some previous submissions won't have a valid answer or a incorrect?

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+  to get an inaccurate picture of the responses.
+
+* If you change the poll so that previous submissions are invalid, those
+  submissions are deleted when learners next view the unit. Those learners are
```
"Those learners..." -> "Learners with invalid submissions are permitted..."

catong commented on en_us/shared/exercises_tools/poll_question.rst:
```diff
@@ -0,0 +1,251 @@
+
+* If you change the poll so that previous submissions are invalid, those
+  submissions are deleted when learners next view the unit. Those learners are
+  permitted to submit new responses. However, those learners do not lose credit
```
"Learners with invalid submissions are permitted to submit new responses, and they do not lose credit in their course progress."

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+Survey Tool
+###################
+
+This section describes how to embed surveys in your course. 
```
"embed" -> "include"

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+Overview 
+*********
+
+You can embed a survey in your course. A survey has multiple answers and
```
Wording is awkward: "multiple answers and multiple questions"?
Also less awkward if plural:
"You can include surveys in your course to collect learner responses to multiple questions."

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+Enable the Survey Tool
+*******************************************
+
+Before you can add a survey to your course, you must enable it in Studio or
```
"survey" -> "surveys"
"enable it" -> "enable the survey tool"

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+Enable the Survey Tool in OLX
+======================================
+
+To enable surveys in your course, you edit the XML file that defines
```
"enable surveys" -> "enable the survey tool"

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+Open the XML file for the course in the ``course`` directory. In the ``course``
+element's ``advanced-modules`` attribute, add the string ``survey``.
+
+For example, the following XML code enables surveys in a course.
```
"enables surveys" -> "enables the survey tool"

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+
+  <course advanced_modules="[&quot;survey&quot;, 
+      &quot;poll&quot;]" display_name="Sample Course" 
+      start="2014-01-01T00:00:00Z">
```
This is just an example, but might as well make it up to date. Change date to "2015" at least

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+Add a Survey in edX Studio
+***************************
+
+Ensure you enable the survey tool before you add the component.
```
"Ensure you enable" -> "You must enable".
OR: "Before you can add a survey component in your course, you must enable the survey tool."
(And include link to the "enable" topics)

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+
+#. From the **Settings** menu, select **Advanced Settings**.
+
+#. In the **Advanced Module List** field, place your cursor between the braces,
```
"between the braces" -> "between the supplied pair of square brackets"

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+    :alt: Advanced modules setting for surveys
+    :width: 400
+
+#. At the bottom of the page, select **Save Changes**.
```
I'd strike "At the bottom of the page." "Select **Save Changes**."

catong replied:
I think several of these procedures have the step result:
"Studio reformats the policy key you just entered to indent it on a new line."

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+   
+#. In the **Display Name** field, enter the name for the component.
+
+#. In the **Feedback** field, enter text that learners see after submitting
```
"after submitting answers." -> "after they submit responses."

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+
+#. In the **Private Results** field, to hide survey results from learners,
+   select **True**. If you leave the default value, **False**, learners see
+   survey results after submitting answers.
```
"after submitting answers." -> "after they submit responses."

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+   select **True**. If you leave the default value, **False**, learners see
+   survey results after submitting answers.
+
+#. In the **Maximum Submissions** field, to allow learners to submit answers
```
Similar to comments for the poll tool, "answers" sounds strange even though it might be what we call them from the Dev point of view. 
I suggest "responses"

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+   submissions.
+
+   .. note:: 
+    If you allow learners to submit answers more than once, you should set
```
"answers" -> "responses"

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+
+   .. note:: 
+    If you allow learners to submit answers more than once, you should set
+    **Private Results** to **True**.
```
Same question as for Poll section -- can you explain why this is recommended?

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+    **Private Results** to **True**.
+
+#. Configure answers for the survey. Each answer is displayed to learners as a
+   column, which a radio button they can select. The answers apply to all
```
Is this supposed to be "with a radio button they can select"?

catong replied:
"The answers apply to all survey questions." I don't understand what is meant.

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+   column, which a radio button they can select. The answers apply to all
+   survey questions.
+
+   #. In each **Answer** field, enter the answer text that learners see as the
```
"enter the text for the column heading that learners will see."

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+   #. In each **Answer** field, enter the answer text that learners see as the
+      column heading.
+
+   #. The survey template contains three answer fields. To add answers, select
```
Same question as for Poll section. It's confusing to see this reference to 3 answer fields after we've already read the first 2 steps previously.

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+   in the left-most column.
+
+   #. You must enter either text or an image path, or both, for each question.
+      To use an image, us the :ref:`Studio URL <File URLs>` for the image.
```
To "enter" an image, "use" (typo "us" -> "use")

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+      the bottom of the list.
+   
+   #. If you use an image, you must enter alternative text in the **Image
+      alternate text** field.
```
Perhaps expand slightly: "you must enter useful alternative text in the Image alternative text field for non-sighted users.
Please check the UI field name -- should be "alternative text" not "alternate text".

catong replied:
Change UI field name to match when/if it is corrected.

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+           
+         Each question must have a value for ``img`` or ``label``, or both.
+     * - ``answers``
+       - An array of answers in the survey. Each answer and answer text.
```
"Each answer and answer text" needs to be a complete sentence.

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+     * - ``max_submissions``
+       - The number of times a learner can submit survey answers.  Use ``0`` to
+         allow unlimited submissions. If you use a value other than ``1``, set
+         ``private_results`` to ``true``.
```
Same question as above -- I don't understand the reason behind this recommendation.

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+Editing Published Surveys
+***************************
+
+Do not publish a survey until you have completed and tested it. You want to avoid changing a survey after learners have begun using it.
```
You want to -> "You should"
Also line wrap

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+
+Do not publish a survey until you have completed and tested it. You want to avoid changing a survey after learners have begun using it.
+
+If you have to edit a survey after learners have submitted answers:
```
"if you have to" -> "If you must"
"have submitted answers:" -> "have submitted answers, take into account the following implications."

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+
+If you have to edit a survey after learners have submitted answers:
+
+* If you edit the value a question or answer, previous submissions are
```
"the value *of*"

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+If you have to edit a survey after learners have submitted answers:
+
+* If you edit the value a question or answer, previous submissions are
+  associated with the new question or answer value. This change can cause you
```
Not clear what is meant by "previous submissions are associated with..."
Reword?

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+
+* If you edit the value a question or answer, previous submissions are
+  associated with the new question or answer value. This change can cause you
+  to get an inaccurate picture of the responses.
```
"can cause you to get" - > "can result in"

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+  associated with the new question or answer value. This change can cause you
+  to get an inaccurate picture of the responses.
+
+* If you change the survey so that previous submissions are invalid, those
```
"previous submissions are invalid." -> maybe something like: 
"existing submissions have answers that no longer correspond to current answer values"? if this is what you are trying to say.

catong commented on en_us/shared/exercises_tools/survey.rst:
```diff
@@ -0,0 +1,256 @@
+
+* If you change the survey so that previous submissions are invalid, those
+  submissions are deleted when learners next view the unit. Those learners are
+  permitted to submit new responses. However, those learners do not lose credit
```
"Learners with invalid submissions are permitted to submit new responses, and they do not lose credit in their course progress."

catong commented:
@mhoeber I'm done with an editing pass. The "question" and "answer' terminology is confusing for both polls and surveys. Perhaps it might help to define what a question and an answer is in the context of each tool, in the overview?

lamagnifica commented:
@catong could you make an explicit list of what remains to get this ready for your thumbs?

mhoeber commented:
@lamagnifica @catong   I just rebased this, changed dates to this wedn.  Affects partner/open edx course staff and OLX.

If you have time for more review that's great, don't think it's a top priority.

lamagnifica commented:
@mhoeber no need for any other reviewers?

mhoeber commented:
@lamagnifica  no been over with with contributors

catong commented:
@mhoeber @lamagnifica None of my comments should hold up this PR. I was confused by the "question" and "answer" terminology but I get it now.
